### ISSUE TITLE
[Feature] Move away from TIFF

### ISSUE THREAD

**Issue by PolarBean** [brainglobe/brainglobe-atlasapi#426]
I wanted to return to this discussion before we embark on our project to integrate many atlases. Currently the main tool for interacting with atlases is ITK-SNAP (its been cited over 9000 times).  It allows you to open templates and overlay segmentations, including region names and colours if a metadata file is provided. It supports a variety of file types with the most popular being Nifti and NRRD but It doesnt support tiff. Now OME-Zarr also seems to be gaining popularity but is not yet supported by ITK-SNAP (I've opened an issue [here](https://github.com/pyushkevich/itksnap/issues/150)). 

I know the API just returns an array and so the file format might not seem to matter, but there are some instances where i think its important. I would at some point like to build a simple website for downloading the atlases currently in brainglobe. It may not be practical to have a backend which converts these on the fly (as we incorporate larger atlases we may run into ram issues). Instead it would be useful just to redirect users to a download link of a hosted atlas.

Perhaps we can compare the pros and cons of each format here and decide which direction we will settle on. 

[NRRD](https://teem.sourceforge.net/nrrd/format.html) is supposed to be an image format for n-dimensional data. It supports some basic metadata in the header which can be read by viewers such as origin and voxel size. it can also store time dimensions. 

[NIFTI](https://brainder.org/2012/09/23/the-nifti-file-format/) designed specifically for neuroimaging data allows for origin, voxel size, affine orientation. It can also store time dimensions and multiple modalities (Though when these features are used it can make the volumes quite large). 

[TIFF](https://en.wikipedia.org/wiki/TIFF) originally designed for office scanners is an imaging format which allows for many tags to be set. A popular extension of TIFF for microscopy data is [OME-TIFF](https://ome-model.readthedocs.io/en/stable/ome-tiff/specification.html). It is not clear to me whether either of these support origin, voxel size, etc, and whether these fields are read by viewers. 

[OME-Zarr](https://link.springer.com/article/10.1007/s00418-023-02209-1) OME-Zarr is an extension of the widely used Zarr format. My understanding is that Zarr is a chunked file format and isnt actually a single file, but rather a directory of files. It is similar to other pyramid file formats and is designed for extremely large datasets that cannot be read into memory. It also enables web viewers such as Neuroglancer to stream the chunks from a repository directly to the viewer. I don't think it's a good choice for us as we have relatively small files which the user will download, not stream on demand. There are more specifications [here](https://ngff.openmicroscopy.org/latest/)

I would lean towards choosing either NRRD or NIFTI due to support for them across tools, as well as their ability to encode spatial metadata. But its up for discussion!

We could compare support across different tools if you think this would help

**Comment by adamltyson** [comment]
I've thought a lot about this, and I'm still not sure there's anything like an ideal format (which is why we haven't made any changes yet). 

TIFF was originally chosen as it's by far the most common format in use by BrainGlobe users, who are predominately microscopists. It can be loaded into microscopy analysis tools (like ImageJ and napari) without downloading any extensions/plugins.

I think there's a general conflict between the two communities:
- Atlas creators who are either from an MRI background, or who are used to ITK-SNAP etc. They would rather NIFTI etc.
- Atlas users, who are microscopists, and are used to ImageJ etc. They would rather TIFF.

There's an argument that it doesn't matter too much because:
- Atlas creators should be confident to convert formats.
- Most users will either use the API or a BrainGlobe tool, and the underlying format doesn't matter.

Re each format:

I would prefer to not use nrrd because I've not seen it used much by either of the above communities, so we would confuse everyone!

NIFTI is a good choice, but things like transforms encoded in the headers have been confusing for users who aren't used to it. It may also not be the best choice for large, multimodal atlases. We don't want a single 30GB file, and if we split them up, it slightly defeats the point of using NIFTI. Another consideration, is how well NIFTI will handle future non-brain atlases.

OME-TIFF is the closest to what we have so far, and is created by many whole-brain microscopes now. It supports lots of metadata, but the ecosystem is far, far less standardised than the NIFTI one in terms of viewer support.

Personally, it seems like Zarr (and by extension OME-Zarr) is likely to be the future for this type of data. It's emerging as a standard in lots of other areas of neuroscience. I do think that a chunked format would be very useful, we've had requests for users wanting to create some kind of BrainGlobe Atlas web API. Zarr has the huge limitation that it is not (yet) a familiar format for users or atlas devs.

To summarise my thoughts - TIFF isn't ideal, but I haven't yet come across a format that is worth the headache of changing everything. Very happy to be convinced otherwise though!

**Comment by PolarBean** [comment]
After our discussion on zoom i'll just summarise some of the points here.

* The format shouldn't matter as people will be accessing through the api
* The desire for ITK-snap functionality is rooted in experience with neuroimaging tools, others with different experience will prefer napari which doesn't natively support nifti or nrrd
* The goal to host atlases in many formats is perhaps not a core brainglobe-atlasapi goal. We could host atlases in many formats and have a frontend for downloading as a parallel initiative which builds on the atlasapi without requiring any core functionality change.
